SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -555,34 +570,47 @@ void DeleteGeneratedSchemaFiles()
 			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Could not clean the generated schema directory '%s'! Please make sure the directory and the files inside are writeable."), *SchemaOutputPath);
 		}
 	}
+}
+
+void CreateSchemaFolder()
+{
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	FString SchemaOutputPath = GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder();
 	PlatformFile.CreateDirectory(*SchemaOutputPath);
 }
 
-void ClearGeneratedSchema()
+void ResetSchemaGeneratorState()
 {
 	ActorClassPathToSchema.Empty();
 	SubobjectClassPathToSchema.Empty();
 	LevelComponentIds.Empty();
 	LevelPathToComponentId.Empty();
 	NextAvailableComponentId = SpatialConstants::STARTING_GENERATED_COMPONENT_ID;
-
-	// As a safety precaution, if the SchemaDatabase.uasset doesn't exist then make sure the schema generated folder is cleared as well.
-	DeleteGeneratedSchemaFiles();
+	SchemaGeneratedClasses.Empty();
 }
 
-bool TryLoadExistingSchemaDatabase()
+bool TryLoadExistingSchemaDatabase(FString FileName /*= ""*/)
 {
-	FFileStatData StatData = FPlatformFileManager::Get().GetPlatformFile().GetStatData(*SchemaDatabaseFileName);
+	if (FileName.IsEmpty())
+	{
+		FileName = SchemaDatabaseFileName;
+	}
+
+	FString RelativeFileName = FPaths::Combine(FPaths::ProjectContentDir(), FileName);
```

Comment:
Maybe an absolute path would be better in terms of what the error message prints?